File: src/main/java/aurocosh/divinefavor/common/potions/base/potion/ModPotionCharge.java
package aurocosh.divinefavor.common.potions.base.potion;

import net.minecraft.client.Minecraft;
import net.minecraft.client.resources.I18n;
import net.minecraft.potion.PotionEffect;

public abstract class ModPotionCharge extends ModPotion {
    public ModPotionCharge(String name, boolean beneficial, int potionColor) {
        super(name, beneficial, potionColor);
    }

    @Override
    public boolean shouldRenderInvText(PotionEffect effect) {
        return false;
    }

    @Override
    public void renderCustomInvText(int x, int y, PotionEffect effect, Minecraft mc) {
        String s1 = I18n.format(getName());

        mc.fontRenderer.drawStringWithShadow(s1, (float) (x + 10 + 18), (float) (y + 6), 16777215);
        String s = "Uses left: " + effect.getAmplifier();
        mc.fontRenderer.drawStringWithShadow(s, (float) (x + 10 + 18), (float) (y + 6 + 10), 8355711);
    }
}

File: src/main/kotlin/aurocosh/divinefavor/common/potions/base/potion/ModPotionCharge.kt
package aurocosh.divinefavor.common.potions.base.potion

import net.minecraft.client.Minecraft
import net.minecraft.client.resources.I18n
import net.minecraft.potion.PotionEffect
import net.minecraftforge.fml.relauncher.Side
import net.minecraftforge.fml.relauncher.SideOnly

abstract class ModPotionCharge(name: String, beneficial: Boolean, potionColor: Int) : ModPotion(name, beneficial, potionColor) {

    override fun shouldRenderInvText(effect: PotionEffect?): Boolean {
        return false
    }

    @SideOnly(Side.CLIENT)
    override fun renderCustomInvText(x: Int, y: Int, effect: PotionEffect, mc: Minecraft) {
        val s1 = I18n.format(name)

        mc.fontRenderer.drawStringWithShadow(s1, (x + 10 + 18).toFloat(), (y + 6).toFloat(), 16777215)
        val s = "Uses left: " + effect.amplifier
        mc.fontRenderer.drawStringWithShadow(s, (x + 10 + 18).toFloat(), (y + 6 + 10).toFloat(), 8355711)
    }
}
